scripts/mkstatic.py

```diff
@@ -604,6 +605,38 @@ def parse(self):
 
         return success
 
+    def make_call_tree(self):
+        '''Create a call tree (list of schemes as they are called) for the specified suite definition file'''
+        success = True
+
+        if not os.path.exists(self._sdf_name):
+            logging.critical("Suite definition file {0} not found.".format(self._sdf_name))
+            success = False
+            return success
+
+        tree = ET.parse(self._sdf_name)
+        suite_xml = tree.getroot()
+        self._name = suite_xml.get('name')
+        # Validate name of suite in XML tag against filename; could be moved to common.py
+        if not (os.path.basename(self._sdf_name) == 'suite_{}.xml'.format(self._name)):
+            logging.critical("Invalid suite name {0} in suite definition file {1}.".format(
+                                                               self._name, self._sdf_name))
+            success = False
+            return success
+
+        # Call tree of all schemes in SDF (with duplicates and subcycles)
```

Comment:
The first 20 lines are identical with the first twenty lines of the `parse` routine. Can this be combined to avoid code duplication?

Reply:
Maybe have an optional (or mandatory) argument `create_call_tree` for the `parse` routine, that switches between what is done in the second half of the `parse` routine?

Reply:
This is a good idea. I initially had the naive idea to try to not modify any existing routines so that I wouldn't have to run as many tests, but it makes more sense the way you suggested. I have implemented this change, let me know how it looks

Reply:
This looks great, thanks for making the change.